app/controllers/users/password_compromised_controller.rb

```diff
@@ -2,18 +2,53 @@
 
 module Users
   class PasswordCompromisedController < ApplicationController
+    include PasswordConcern
     before_action :confirm_two_factor_authenticated
     before_action :verify_feature_toggle_on
 
     def show
-      session.delete(:redirect_to_password_compromised)
-      @after_sign_in_path = after_sign_in_path_for(current_user)
+      session.delete(:redirect_to_change_password)
+      @update_user_password_form = UpdateUserPasswordForm.new(current_user)
+      @forbidden_passwords = forbidden_passwords
       analytics.user_password_compromised_visited
     end
 
+    def update
+      @update_user_password_form = UpdateUserPasswordForm.new(current_user, user_session)
+
+      result = @update_user_password_form.submit(user_password_params)
+      updated_result = result.to_h.merge({ original_password_compromised: true })
```

Comment:
Can we update the `analytics_method` definition to include and document this new parameter?

https://github.com/18F/identity-idp/blob/25e66ea77a99ea4a6d87394dfd453c2f6e0a5509/app/services/analytics_events.rb#L4465-L4471

Also, the curly braces are not required.

```suggestion
      updated_result = result.to_h.merge(original_password_compromised: true)
```

Would it have been possible to assign this value in `UpdateUserPasswordForm#extra_analytics_event` so that we don't have to post-process the result and look for analytics properties in two places?